### ISSUE TITLE
Components: assess `Heading` styles in the context of the recent design updates

### ISSUE THREAD

**Issue by ciampo** [WordPress/gutenberg#35464]
As originally discussed in https://github.com/WordPress/gutenberg/pull/35400#discussion_r724311248, it looks like the `Heading` component may currently not fit the needs of the most recent designs (specifically for the Global Styles sidebar, in this case)

We should assess the situation and, if necessary, decide how to proceed in order to have a component offering the new heading styles needed for the Global Styles sidebar UI.

Examples that came up recently:

- https://github.com/WordPress/gutenberg/pull/35415#discussion_r725881998
- https://github.com/WordPress/gutenberg/pull/35568#issuecomment-942120232
- https://github.com/WordPress/gutenberg/pull/35369#discussion_r729711696

Useful links:

- Component's definition and code: https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/heading
- Storybook example: https://wordpress.github.io/gutenberg/?path=/story/components-experimental-heading--default

**Comment by ciampo** [comment]
@mtias  and @jasmussen , do you have any insights on this?

**Comment by jasmussen** [comment]
In exploring the sidebar/inspector/global styles design that is being finalized in #34574, it seems clear that the uppercase styling is used both for input labels and for some subheadings, so it appears to be more of a style than a component, perhaps somewhat comparable to how we have a help-text style (which is 12px font size, default font weight, default text case). CC: @pablohoneyhoney in case he has nuance to add.

**Comment by ciampo** [comment]
Are you suggesting we introduce variants to the `Heading` component?

In general, I think that if we want the `Heading` component to be used for all headings, we may need to:

- define clearly all of the allowed "heading" styles
- update the `Heading` component to support these styles
- refactor the code for all headings to use the `Heading` component

Another example of where a heading is required (but doesn't match necessarily the "heading" styles) is `ToolsPanel` (see related conversation https://github.com/WordPress/gutenberg/pull/35415#discussion_r725881998)

**Comment by jasmussen** [comment]
I'm not necessarily suggesting anything in specific here, only providing context to help a path forward. It's possible that it could exist purely as a heading level style (such as the H3 style), but I do suspect that there might be contexts where the subheading style needs a higher heading level, but still the visually lower styling (such as when showing the contents of a particular panel in a popover opened by the block toolbar).

**Comment by ciampo** [comment]
Absolutely, I think we should definitely be able to decouple the HTML semantic (e.g. `h1/h2/h3/h4/h5/h6`) from the visual style

**Comment by jasmussen** [comment]
I also think we want to end up with very few styles overall, so as to find a good balance, and obvious implied usage guidelines.

**Comment by ciampo** [comment]
Another example where updated Heading styles would have turned useful (thank @jasmussen) https://github.com/WordPress/gutenberg/pull/35568#issuecomment-942120232

**Comment by ciampo** [comment]
A few more examples:

- the `ColorGradient` and `ColorPalette` headings added in #35970
- https://github.com/WordPress/gutenberg/pull/35783#discussion_r736744350

**Comment by ciampo** [comment]
See also the `Subtitle` heading discussed in https://github.com/WordPress/gutenberg/pull/36718#issuecomment-975391614

**Comment by mirka** [comment]
I think we're good here, with the new heading styles we have in `@wordpress/ui` `Text`?
